### ISSUE TITLE
read ansible.cfg from project/current directory

### ISSUE THREAD

**Issue by foto-andreas** [sadv1r/ansible-vault-editor-idea-plugin#302]
Hi,

is this only a forgotten TODO or did you encounter problems implementing the usage of ansible.config in current directory?

```
64 | //TODO ansible.cfg (in the current directory)
65 |  
66 | // in the home directory
67 | getPasswordFilePathFromConfig(SystemProperties.getUserHome() + "/.ansible.cfg")
```

Andreas

**Comment by foto-andreas** [comment]
Adding my idea:
```
    // in project directory
    for (p in ProjectManager.getInstance().openProjects) {
        for (r in ProjectRootManager.getInstance(p).contentRoots) {
            getPasswordFilePathFromConfig(r.path + "/ansible.cfg")
                ?.let { return r.path + "/" + it }
        }
        break;
    }

    // in current directory
    getPasswordFilePathFromConfig("ansible.cfg")
        ?.let { return File(it).absolutePath }
```

Beware. I have no idea on the correct IJ-APIs to use, but this solution works for me.

Open: 
* using projects[0] - is this correct when multiple projects are open. Should use the current project
* is searching all content roots ok?
* ...

Andreas

**Comment by foto-andreas** [comment]
working on pull request #304 on it

**Comment by Grogdunn** [comment]
In our company we use to have all ansible files under subdirectory, this PR can support this?
Eg:
```
[root]/ansible/ansible.cfg
or
[root]/infrastructure/ansible/ansible.cfg
```

**Comment by foto-andreas** [comment]
not currently, only if the vaulted files are also in that directory. how will ansible-playbook find these files when called?

**Comment by Grogdunn** [comment]
Simply we have a Makefile with a target like this:
```
deploy: build
	ANSIBLE_CONFIG=ansible/ansible.cfg ansible-playbook \
        --private-key ~/.ssh/superkey.pem \
        -i ansible/inventories/prod/hosts \
        --user admin \
        ansible/main.yml
```
or simple cd and execute the playbook

**Comment by Grogdunn** [comment]
How to configure IDEA with ANSIBLE_CONFIG? I don't know.

**Comment by foto-andreas** [comment]
For complex setups: https://plugins.jetbrains.com/plugin/14353-ansible-vault-integration/tutorials/vault-file-as-script

**Comment by sadv1r** [comment]
> How to configure IDEA with ANSIBLE_CONFIG? I don't know.

Hi 👋!

You can set an environment variable on the system level, like for the default tool.

**Comment by sadv1r** [comment]
@foto-andreas thanks for the issue, I really forgot about that todo but as I can see in the comments now, there are some cases when people need to read the file from different locations, like custom subdirectory.

I'm trying to stick with the default locations from [Ansible documentation](https://docs.ansible.com/ansible/latest/reference_appendices/config.html#the-configuration-file), but maybe you already can make this work by using password file with a script, where you can do literally anything

**Comment by Grogdunn** [comment]
> > How to configure IDEA with ANSIBLE_CONFIG? I don't know.
> 
> Hi 👋!
> 
> You can set an [environment variable](https://github.com/sadv1r/ansible-vault-editor-idea-plugin/blob/main/src/main/kotlin/ru/sadv1r/idea/plugin/ansible/vault/editor/AnsibleConfigReader.kt#L58) on the system level, like for the default tool.

Yes and no... Because I need to set multiple time environment variable (that is global) once for each project I open, and sometimes I open 2 to 4 projects, at same time, all with ansible inside (and a lot of other things), and sometimes ansible.cfg files are under a path or sometimes under some "funky" path. And cherry on the cacke, each project has a different vault password file.
So I can specify the ansible.cfg file position per project or can be done a sort of a discovery on project paths/index.

What do you think?

**Comment by fede843** [comment]
Hi, is there any progress or update on this one? I am an old user of this plugin, but currently I am re organizing some projects. It'd come very andy to get the `vault_password_file` from the `ansible.cfg`. In my case I use a link in each root project, because have more than one configuration going on.
